fetch_index.py

```diff
@@ -0,0 +1,354 @@
+                return []
+            try:
+                response_data = response.json()
+                response_data = response_data.get('d', {}).get('results')
+            except Exception as exception:
+                logger.error('Error while parsing the get list response for site: {} from url: {}. Error: {}'.format(
+                    site, rel_url, exception))
+                return []
+            logger.info(
+                'Successfuly fetched and parsed the list response for site: {} from SharePoint'.format(site))
+
+            for index in range(len(response_data)):
+                response_data[index] = dict((k.lower(), v)
+                                            for k, v in response_data[index].items())
+                response_data[index].pop('__metadata')
+
+            if index:
+                json_object = json.dumps(response_data, indent=4)
+                logger.info(
+                    'Indexing the list for site: {} to the Workplace'.format(site))
+                try:
+                    self.ws_client.index_documents(
+                        http_auth=self.ws_token,
+                        content_source_id=self.ws_source,
+                        documents=json_object
+                    )
+                    logger.info(
+                        'Successfully indexed the list for site: {} to the workplace'.format(site))
+                except Exception as exception:
+                    logger.error('Error while indexing the list for site: {} to the workplace. Error: {}'.format(
+                        site, exception))
+                    self.is_error = True
+                    return []
+            responses.append(response_data)
+        return responses
+
+    def get_lists_paths(self, sites, response_data=None):
+        logger.info('Extracting list name')
+        lists = {}
+        if not response_data:
+            logger.info(
+                'List response is not present. Fetching the list for sites')
+            response_data = self.index_lists(sites, index=False) or []
+
+        lists = {}
+        for response in response_data:
+            for result in response:
+                lists[result.get('title')] = result.get('parentweburl')
+
+        return lists
+
+    def index_items(self, lists, index):
+        responses = []
+        logger.info('Fetching all the items for the lists')
+        for list_content, value in lists.items():
+            rel_url = urljoin(
+                self.sharepoint_host, "{}/_api/web/lists/getbytitle(\'{}\')/items".format(list_content, value))
+
+            logger.info(
+                'Fetching the items for list: {} from url: {}'.format(list, rel_url))
+
+            # Condition to check if the list does not have a file/folder
+            if self.sharepoint_client.get(rel_url, "?$select=Folder Eq null") or self.sharepoint_client.get(rel_url, "?$select=File Eq null"):
```

Comment:
Is there a benefit for checking whether there are any folders/files before trying to fetch them directly?
Could we simplify the code and just query for files/folder?

Reply:
Thanks, @ioanatia.  I've tried to resolve all the issues listed above and will fix the folder/files issue in the upcoming PR update.